# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1355061731 -3600
#      Sun Dec 09 15:02:11 2012 +0100
# Node ID b0ad083e19583f8dc23fd5965cec5b796b2f43d0
# Parent  d90b426b99881584934c80d86acd352cca4718a4
respect charset when updating the history widget

* history-dockwidget.cc: using a QString converted from local8Bit for
  updating the history widget in update_history_callback ()

diff --git a/libgui/src/history-dockwidget.cc b/libgui/src/history-dockwidget.cc
--- a/libgui/src/history-dockwidget.cc
+++ b/libgui/src/history-dockwidget.cc
@@ -195,17 +195,17 @@ history_dock_widget::update_history_call
 
       _history_model->insertRows (clientHistoryLength, elts_to_add);
 
       for (int i = clientHistoryLength; i < serverHistoryLength; i++)
         {
           std::string entry = command_history::get_entry (i);
 
           _history_model->setData (_history_model->index (i),
-                                   QString::fromStdString (entry));
+                      QString::fromLocal8Bit (entry.data (), entry.size ()));
         }
 
       // FIXME -- does this behavior make sense?  Calling
       // _history_list_view->scrollToBottom () here doesn't seem to
       // have any effect.  Instead, we need to request that action
       // and wait until the next event occurs in which no items
       // are added to the history list.
 
